scripts/upgrade.sh

```diff
@@ -227,10 +227,12 @@ compare_versions() {
     return 0
 }
 
-# Function to convert "2023-06-20T12:30:15Z" format to "2023-06-20 12:30:15"
+# Function to convert "2023-06-20T12:30:15Z" format to "2023-06-01 00:00:00"
+# Will not consider anything different but month and year
+# Every date will be set to day 01 at 00:00:00
 convert_date_format() {
     date_str=$1
-    echo "$date_str" | sed 's/T/ /; s/Z//'
+    echo "$date_str" | sed -E 's/^([0-9]{4}-[0-9]{2})-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z/\1-01 00:00:00/'
```

Comment:
Do we need to first affirm the date coming in is valid before converting?

Reply:
@osodracnai is going off shift so I'm picking this up, and have reworked the logic a bit.